src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -88,6 +98,11 @@ public Person toModelType() throws IllegalValueException {
             personTags.add(tag.toModelType());
         }
 
+        final Map<Exam, Score> personExamScores = new HashMap<>();
+        for (JsonAdaptedExamScore examScore : examScores) {
+            personExamScores.put(examScore.toModelTypeExam(), examScore.toModelTypeScore());
```

Comment:
nice job bringing the scores into a map to simplify the subsequent creation of the corresponding person.